internal/mode/static/state/graph/backend_tls_policy_test.go

```diff
@@ -0,0 +1,344 @@
+						Hostname:   "foo.test.com",
+					},
+				},
+			},
+			isValid: false,
+		},
+		{
+			name: "invalid ca cert ref kind",
+			tlsPolicy: &v1alpha2.BackendTLSPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "tls-policy",
+					Namespace: "test",
+				},
+				Spec: v1alpha2.BackendTLSPolicySpec{
+					TargetRef: *targetRefNormalCase,
+					TLS: v1alpha2.BackendTLSPolicyConfig{
+						CACertRefs: localObjectRefInvalidKind,
+						Hostname:   "foo.test.com",
+					},
+				},
+			},
+			isValid: false,
+		},
+		{
+			name: "invalid ca cert ref group",
+			tlsPolicy: &v1alpha2.BackendTLSPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "tls-policy",
+					Namespace: "test",
+				},
+				Spec: v1alpha2.BackendTLSPolicySpec{
+					TargetRef: *targetRefNormalCase,
+					TLS: v1alpha2.BackendTLSPolicyConfig{
+						CACertRefs: localObjectRefInvalidGroup,
+						Hostname:   "foo.test.com",
+					},
+				},
+			},
+			isValid: false,
+		},
+		{
+			name: "invalid case with well known certs",
+			tlsPolicy: &v1alpha2.BackendTLSPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "tls-policy",
+					Namespace: "test",
+				},
+				Spec: v1alpha2.BackendTLSPolicySpec{
+					TargetRef: *targetRefNormalCase,
+					TLS: v1alpha2.BackendTLSPolicyConfig{
+						WellKnownCACerts: (helpers.GetPointer(v1alpha2.WellKnownCACertType("unknown"))),
+						Hostname:         "foo.test.com",
+					},
+				},
+			},
+			isValid: false,
+		},
+		{
+			name: "invalid case neither TLS config option chosen",
+			tlsPolicy: &v1alpha2.BackendTLSPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "tls-policy",
+					Namespace: "test",
+				},
+				Spec: v1alpha2.BackendTLSPolicySpec{
+					TargetRef: *targetRefNormalCase,
+					TLS: v1alpha2.BackendTLSPolicyConfig{
+						Hostname: "foo.test.com",
+					},
+				},
+			},
+			isValid: false,
+		},
+		{
+			name: "invalid case with too many ca cert refs",
+			tlsPolicy: &v1alpha2.BackendTLSPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "tls-policy",
+					Namespace: "test",
+				},
+				Spec: v1alpha2.BackendTLSPolicySpec{
+					TargetRef: *targetRefNormalCase,
+					TLS: v1alpha2.BackendTLSPolicyConfig{
+						CACertRefs: localObjectRefTooManyCerts,
+						Hostname:   "foo.test.com",
+					},
+				},
+			},
+			isValid: false,
+		},
+		{
+			name: "invalid case with too many ancestors",
+			tlsPolicy: &v1alpha2.BackendTLSPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "tls-policy",
+					Namespace: "test",
+				},
+				Spec: v1alpha2.BackendTLSPolicySpec{
+					TargetRef: *targetRefNormalCase,
+					TLS: v1alpha2.BackendTLSPolicyConfig{
+						CACertRefs: localObjectRefNormalCase,
+						Hostname:   "foo.test.com",
+					},
+				},
+				Status: v1alpha2.PolicyStatus{
+					Ancestors: ancestors,
+				},
+			},
+			isValid:    false,
+			caCertName: types.NamespacedName{Namespace: "test", Name: "configmap"},
+			ignored:    true,
+		},
+	}
```

Comment:
double checking if we have a missing test case here

Reply:
Sorry, missed this one!!

It's blocked by the validation, so we don't explicitly check for it, but we will take CACertRefs over WellKnownCerts every time, meaning WellKnownCerts will simply be ignored. I'll add an explicit check for it and a corresponding test just in case though.